typescript/agentkit/src/action-providers/baseSqlApi/schemas.ts

```diff
@@ -0,0 +1,15 @@
+import { z } from "zod";
+
+/**
+ * Action schemas for the baseSqlApi action provider.
+ *
+ * This file contains the Zod schemas that define the shape and validation
+ * rules for action parameters in the baseSqlApi action provider.
+ */
+
+export const BaseSqlApiSchema = z.object({
+  /**
+   * The SQL query to run
+   */
+  sqlQuery: z.string().describe("The sql query to execute, using the defined tables and fields"),
+});
```

Comment:
add query limitations
```suggestion
  sqlQuery: z
    .string()
    .min(1, "SQL query cannot be empty")
    .max(10000, "Query exceeds maximum length of 10,000 characters")
    .describe(
      `The SQL query to execute using ClickHouse syntax. Must be a read-only SELECT statement. ` +
      `Limitations: max 10,000 characters, max 5 JOINs, no cartesian products, 30s timeout. API supports a max of 10,000 result rows but limit it to 10 unless otherwise specified.`
    ),});
```